### PR TITLE
Align docs for all effects

### PR DESCRIPTION
**Pull request by patrick-tolosa**
#### What does this PR solve?
Add JSdoc to all effect actions

#### Final checklist
- [X] JSdoc - Add proper docstrings based on our PHP implementation
- [ ] JSdoc - Hide private functions using @private
- [ ] Implementation is aligned to Spec.
- [ ] Tests - Add proper tests to the added code

### CODE REVIEW

File: src/actions/effect/EffectActions/EffectActionWithLevel.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import LeveledEffectAction from "./LeveledEffectAction";
 /**
  * @class EffectActionWithLevel
  * @augments LeveledEffectAction
- * @description A class for all effects that include a level method
+ * @description A class that provides a built in level() method that sets that level of the effect
```
Comment (strausr) [inline src/actions/effect/EffectActions/EffectActionWithLevel.ts:6]: I think this should be "that set the level of the effect"

File: src/actions/effect/assistColorBlind.ts [hunk 2]
```diff
@@ -2,16 +2,32 @@ import Action from "../Action";
 import QualifierValue from "../../qualifier/QualifierValue";
 import Qualifier from "../../qualifier/Qualifier";
 
+/**
+ * @class AssistColorBlindEffectAction
+ * @augments Action
+ *
```
Comment (strausr) [inline src/actions/effect/assistColorBlind.ts:8]: space:)

File: src/actions/effect/cartoonify.ts [hunk 3]
```diff
@@ -13,12 +18,26 @@ class CartoonifyEffect extends Action {
     this.effectName = effectName;
   }
 
+  /**
+   * @description The thickness of the lines. (Range: 0 to 100, Server default: 50)
+   * @param {number} cartoonifyLevel The thickness of the lines
+   * @return {this}
+   */
   strength(cartoonifyLevel: number) {
```
Comment (strausr) [inline src/actions/effect/cartoonify.ts:26]: Should we add return type?

File: src/actions/effect/leveled/loop.ts [hunk 4]
```diff
@@ -1,6 +1,10 @@
 import LeveledEffectAction from "../EffectActions/LeveledEffectAction";
 
-
+/**
+ * @description Delivers a video or animated GIF that contains additional loops of the video/GIF.
+ * @class LoopEffectAction
+ * @augments LeveledEffectAction
```
Comment (strausr) [inline src/actions/effect/leveled/loop.ts:6]: we should have the same convention of @description at the end

File: src/actions/effect/outline.ts [hunk 5]
```diff
@@ -3,6 +3,11 @@ import QualifierValue from "../../qualifier/QualifierValue";
 import Qualifier from "../../qualifier/Qualifier";
 import {prepareColor} from "../../utils/prepareColor";
 
+/**
+ * @description Adds an outline to a transparent image. For examples, see the Image Transformations guide.
```
Comment (strausr) [inline src/actions/effect/outline.ts:7]: move to end?

File: src/actions/effect/outline.ts [hunk 6]
```diff
@@ -26,6 +32,7 @@ class Outline extends Action {
   /**
    * The thickness of the outline in pixels. (Range: 1 to 100, Server default: 5)
    * @param {number} width
+   * @return {this}
    */
   width(width?:number | string) {
```
Comment (strausr) [inline src/actions/effect/outline.ts:37]: return type?

**Review by strausr — CHANGES_REQUESTED** [review]
Look good the only thing is that in certain places the description is at the beginning of the dosctring and it might be a good idea to keep the same convention and have it at the end. Also, would consider adding return type to internal functions since they were added to the jsdocs

File: src/actions/effect/EffectActions/EffectActionWithLevel.ts (hunk 1 above)
Comment (patrick-tolosa) [reply src/actions/effect/EffectActions/EffectActionWithLevel.ts:6]: I'm not sure what you mean ,can you write the entire docstring?

File: src/actions/effect/cartoonify.ts (hunk 3 above)
Comment (patrick-tolosa) [reply src/actions/effect/cartoonify.ts:26]: added

File: src/actions/effect/outline.ts (hunk 6 above)
Comment (patrick-tolosa) [reply src/actions/effect/outline.ts:37]: added

File: src/actions/effect/outline.ts (hunk 5 above)
Comment (patrick-tolosa) [reply src/actions/effect/outline.ts:7]: I'm not sure, I think for functions it makes more sense for it to be first

Maybe it should always be first, even in classes?

File: src/actions/effect/assistColorBlind.ts (hunk 2 above)
Comment (patrick-tolosa) [reply src/actions/effect/assistColorBlind.ts:8]: Removed

File: src/actions/effect/EffectActions/EffectActionWithLevel.ts (hunk 1 above)
Comment (strausr) [reply src/actions/effect/EffectActions/EffectActionWithLevel.ts:6]: A class that provides a built in level() method that sets the level of the effect

File: src/actions/effect/leveled/loop.ts (hunk 4 above)
Comment (patrick-tolosa) [reply src/actions/effect/leveled/loop.ts:5]: As discussed

**Review by strausr — APPROVED** [review]

**Pull request merged**